docs/architecture/adr-061-core-module-api.md

```diff
@@ -0,0 +1,490 @@
+# ADR 061: Core Module API
+
+## Changelog
+
+* 2022-08-18 First Draft
+
+## Status
+
+PROPOSED Not Implemented
+
+## Abstract
+
+A new core API is proposed to replace the existing `AppModule` and `sdk.Context` frameworks a set of core services
+and a new `Handler` struct. This core API aims to:
+- be simpler,
+- more extensible
+- and more stable than the current framework
+while also
+- enabling deterministic events and queries, and
+- supporting event listeners
+- and [ADR 033: Protobuf-based Inter-Module Communication](./adr-033-protobuf-inter-module-comm.md) clients.
+
+## Context
+
+Historically modules have exposed their functionality to the state machine via the `AppModule` and `AppModuleBasic`
+interfaces which have the following shortcomings:
+* both `AppModule` and `AppModuleBasic` need to be defined which is counter-intuitive
+* apps need to implement the full interfaces, even parts they don't need
```

Comment:
Not technically, as they can define no-ops for many of them, which they do.

Reply:
Perhaps then the con is: "Modules are required to supply unnecessary boilerplate to implement no-op"

Reply:
Although, I _think_ some of this could be addressed with breaking refactors to `AppModule`, `AppModuleBasic`.  That is, if [AppModuleBasic](https://github.com/cosmos/cosmos-sdk/blob/249643325182a85dc60d2baae7d867b1db765467/types/module/module.go#L48-L60) was decomposed into domain-specific extension interfaces.